eng/pipelines/coreclr/templates/build-corelib-job.yml

```diff
@@ -0,0 +1,124 @@
+    - name: stripSymbolsArg
+      value: ''
+    # Strip symbols only on the release build
+    - ${{ if eq(parameters.buildConfig, 'Release') }}:
+      - name: stripSymbolsArg
+        value: '-stripsymbols'
+    - name: compilerArg
+      value: ''
+    - name: publishLogsArtifactPrefix
+      value: 'BuildLogs_CoreLib'
+    - ${{ if and(ne(variables['System.TeamProject'], 'public'), ne(variables['Build.Reason'], 'PullRequest')) }}:
+      # Variables used to publish packages to blob feed
+      - name: dotnetfeedUrl
+        value: https://dotnetfeed.blob.core.windows.net/dotnet-coreclr/index.json
+      - name: dotnetfeedPAT
+        value: $(dotnetfeed-storage-access-key-1)
+      # Variables used by arcade to gather asset manifests
+      - name: _DotNetPublishToBlobFeed
+        value: true
+    - name: officialBuildIdArg
+      value: ''
+    - ${{ if and(eq(variables['System.TeamProject'], 'internal'), ne(variables['Build.Reason'], 'PullRequest')) }}:
+      - name: officialBuildIdArg
+        value: '/p:OfficialBuildId=$(Build.BuildNumber)'
+    - name: enforcePgoArg
+      value: ''
+    # The EnforcePGO script is only supported on Windows and is not supported on arm or arm64.
+    - ${{ if and(eq(parameters.buildConfig, 'Release'), and(eq(parameters.osGroup, 'Windows_NT'), not(or(eq(parameters.archType, 'arm64'), eq(parameters.archType, 'arm'))))) }}:
+      - name: enforcePgoArg
+        value: '-enforcepgo'
+
+    - ${{ parameters.variables }}
+
+    steps:
+
+    # Install native dependencies
+    # Linux builds use docker images with dependencies preinstalled,
+    # and FreeBSD builds use a build agent with dependencies
+    # preinstalled, so we only need this step for OSX and Windows.
+    - ${{ if eq(parameters.osGroup, 'OSX') }}:
+      - script: sh $(Build.SourcesDirectory)/eng/install-native-dependencies.sh $(osGroup)
+        displayName: Install native dependencies
+    - ${{ if eq(parameters.osGroup, 'Windows_NT') }}:
+      # Necessary to install python
+      - script: $(Build.SourcesDirectory)\eng\common\init-tools-native.cmd -InstallDirectory $(Build.SourcesDirectory)\native-tools -Force
+        displayName: Install native dependencies
+
+    # # Install internal tools on official builds
+    # # Since our internal tools are behind an authenticated feed,
+    # # we need to use the DotNetCli AzDO task to restore from the feed using a service connection.
+    # # We can't do this from within the build, so we need to do this as a separate step.
+    - ${{ if and(eq(variables['System.TeamProject'], 'internal'), ne(variables['Build.Reason'], 'PullRequest')) }}:
+      - template: /eng/pipelines/common/restore-internal-tools.yml
+        parameters:
+          installDotnet: true
+
+    # Build Private CoreLib
+    - script: $(Build.SourcesDirectory)$(dir)build$(scriptExt) -subsetCategory coreclr -subset corelib $(crossArg) -arch $(archType) -c $(buildConfig) $(officialBuildIdArg) -ci
```

Comment:
Do we need the crossgened corelib as well? We usually run tests against nativecorelib. @jkoritzinsky ?

Reply:
We'll crossgen CoreLib in the CoreCLR build job. The libraries build will get the IL CoreLib when compiling but the crossgened one when running tests.

Reply:
I see. Does it make sense to build corelib twice? Doesn't it make more sense to just crossgen corelib here and use this job + the runtime job to run tests?

Reply:
Crossgen isn't built here, so we can't run it.

If we want to just build it twice that's another option.

Reply:
I think we should keep it as simple as possible as part of this PR so I'm fine with building the managed corelib here and then the native in Build CoreCLR legs and use those for testing as we're currently doing. Then we can improve the builds if needed.

Reply:
yeah agree, lets get the parallelization working and then we can optimize things further as needed.